### PR TITLE
Add Thai W2P

### PR DESCRIPTION
**Pull request by wannaphong**
### What does this changes

I add Thai Word-to-Phoneme converter to `pythainlp.transliterate.pronunciate`. It's converter thai word to thai phoneme.

GitHub: https://github.com/wannaphong/Thai_W2P

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [X] Passed code styles and structures
- [x] Passed code linting checks and unit test

### CODE REVIEW

**Comment by wannaphong** [comment]
Google colab for test https://colab.research.google.com/drive/1MYc5PztkY26-O1bgRXmh9rShx0GLwFWp?usp=sharing

**Comment by wannaphong** [comment]
# Model Card

**Model Details**

- Developer: Wannaphong Phatthiyaphaibun
- Model date: 2020-12-29
- Model version: 0.1
- Used in PyThaiNLP version: __
- GitHub:  ___
- License: CC0
- train notebook: https://github.com/wannaphong/Thai_W2P/blob/main/train.ipynb

**Intended Use**

- Converter thai word to thai phoneme
- Not suitable for other language.

**Factors**

- Based on thai word to thai phoneme problems.

**Metrics**

- Evaluation metrics include phoneme error rate (number error / number phonemes)

**Training Data**

Thai W2P

**Evaluation Data** 

Thai W2P

**Quantitative Analyses**

```
epoch: 100
step: 100, loss: 0.03179970383644104
step: 200, loss: 0.04126007482409477
step: 300, loss: 0.01877519115805626
step: 400, loss: 0.03311225399374962
per: 0.0432
per: 0.0419
```

**Ethical Considerations**

thai phoneme based on website (wiktionary, Royal Institute et cetera). It may not be the dialect that you use in everyday life.

**Caveats and Recommendations**

- 1 Thai word only

File: pythainlp/transliterate/core.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def transliterate(
     :Options for engines:
         * *icu* - International Components for Unicode (ICU)
         * *ipa* - International Phonetic Alphabet (IPA) by epitran
+        * *w2p* - Thai Word-to-Phoneme converter.
+          It's converter thai word to thai phoneme.
```
Comment (bact) [inline pythainlp/transliterate/core.py]: For me, it's a bit confusing when "w2p" and "thaig2p" said that their output are both "phoneme" but they looks different. Does it possible to find a more specific name?

```python
        transliterate("ภาพยนตร์", engine="thaig2p")
        # output:'pʰ aː p̚ ˥˩ . pʰ a ˦˥ . j o n ˧'

        transliterate("ภาพยนตร์", engine="w2p")
        # output: 'พาบ-พะ-ยน'
```

**Review by bact — COMMENTED** [review]
Does it possible now for the transliterate() function to convert multiple Thai words into their sound representation?

If not, I suggest that it should. Because that's how other engine do. We should keep the types of input and output consistent.

This can be done by tokenized the input first, then do the w2p for each token.

File: pythainlp/transliterate/core.py (hunk 1 above)
Comment (wannaphong) [reply pythainlp/transliterate/core.py]: For convert multiple Thai words, It needs word segmentation. I think the user should choose word segmentation himself.

Comment (wannaphong) [reply pythainlp/transliterate/core.py]: > 
> 
> For me, it's a bit confusing when "w2p" and "thaig2p" said that their output are both "phoneme" but they looks different. Does it possible to find a more specific name?
> 
> ```python
>         transliterate("ภาพยนตร์", engine="thaig2p")
>         # output:'pʰ aː p̚ ˥˩ . pʰ a ˦˥ . j o n ˧'
> 
>         transliterate("ภาพยนตร์", engine="w2p")
>         # output: 'พาบ-พะ-ยน'
> ```

I agree.

Comment (wannaphong) [reply pythainlp/transliterate/core.py]: > 
> 
> > For me, it's a bit confusing when "w2p" and "thaig2p" said that their output are both "phoneme" but they looks different. Does it possible to find a more specific name?
> > ```python
> >         transliterate("ภาพยนตร์", engine="thaig2p")
> >         # output:'pʰ aː p̚ ˥˩ . pʰ a ˦˥ . j o n ˧'
> > 
> >         transliterate("ภาพยนตร์", engine="w2p")
> >         # output: 'พาบ-พะ-ยน'
> > ```
> 
> I agree.

Change w2p to pronunciate

**Comment by wannaphong** [comment]
I changed `w2p` from `pythainlp.transliterate.transliterate` to `pythainlp.transliterate.pronunciate`.

**Comment by bact** [comment]
I think we're good to go. Thanks @wannaphong

**Comment by bact** [comment]
All Thai G2P, Thai W2P, etc. can be converted to the singleton pattern but we can do that later.

**Comment by wannaphong** [comment]
@bact Thank you.

**Pull request merged**